src/Storage.gs

```diff
@@ -0,0 +1,115 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Contains classes used to persist data and access it.
+ */
+
+/**
+ * Creates a new storage instance.
+ * @param {string} prefix The prefix to use for keys in the properties and cache.
+ * @param {PropertiesService.Properties} properties The properties instance to use.
+ * @param {CacheService.Cache} opt_cache The optional cache instance to use.
+ * @constructor
+ */
+function Storage(prefix, properties, opt_cache) {
+  this.prefix_ = prefix;
+  this.properties_ = properties;
+  this.cache_ = opt_cache;
+  this.memory_ = {};
+}
+
+/**
+ * The TTL for cache entries, in seconds.
+ * @type {number}
+ * @private
+ */
+Storage.CACHE_EXPIRATION_TIME_SECONDS = 21600;
```

Comment:
A comment would be nice.
`Storage.CACHE_EXPIRATION_TIME_SECONDS = 21600; // 6 hours`

Reply:
Done